doc/database/multi_namespace_db_instances.md

```diff
@@ -0,0 +1,561 @@
+
+*  Global services like database, snmp, pmon, telemetry running in the dockers running in the linux "host" namespace.
+   We call it "global" namespace.
+	
+*  NPU specific services like database, swss, syncd, bgp, teamd, lldp etc which runs in a separate "namespace" created. The method used currently to seggregate services per NPU is "linux network namespace" and there is a one-to-one mapping between the number of NPUs and linux network namesapces created. We call it "NPU" namespace.
+
+The database docker in the "global" namespace can be called the "global DB" service. The redis databases available here (decided by contents of database_config.json) would be APPL_DB, CONFIG_DB used to store the system wide attributes like AAA, syslog, ASIC to interface name mapping etc. 
+
+There are also database docker in the "NPU" namespace. The redis databases there will include all the DB's viz. APPL_DB, CONFIG_DB, ASIC_DB, COUNTERS_DB etc and is used to store the interface/counters/state etc specific to the interfaces/features on that NPU.
+
+This approach is backward compatible to the Sonic devices with single NPU. The "globalDB" service could be considered equivalent to the database docker started in the linux "host" network namespace in single NPU device. We will discuss more on the design details of database docker service in this document.
+
+There will be a config_db.json file per database service, it will be named "/etc/sonic/config_db.json" for the "global DB" and /etc/sonic/config_db{NS}.json for the database service in the {NS} namespace.
+
+## Enhancements to Multi-DB design to support Multiple namespaces
+
+The file database_config.json contains the startup configuration which dictates the redis server host/port/unix_socket configurations + various available databases ( eg: APP_DB, CONFIG_DB ) etc. With the introduction of multiple namespaces, there is a need for separate database_config.json per namespace. This config file is created in the "working redis directories" during the database docker service startup. 
+
+The database service for a NPU linux namespace {NS} will use "/var/run/redis{NS}" as the "working redis directory" to create the various files like redis.sock, database_config.json etc. The "globalDB" database service would use "/var/run/redis" as the working directory.
+	
+![center](./img/redis.png)
+
+Following are the major design changes
+
+* A new file **databases.json** is introduced. It will contain the details of the namespaces and the corresponsing database_config.json files. This file would be created by the "globalDB" service in the directory "/var/run/redis/sonic-db/". In the below example, we consider a SONIC device with 3 NPUS's and hence have 3 namespaces refered with namespaceID "0", "1" & "2".
+
+```json
+[
+    {
+        "namespace" : "global",
+        "config" : "../redis/sonic-db/database_config.json"
+    },
+    {
+        "namespace" : "0",
```

Comment:
>"namespace" : "0", [](start = 8, length = 18)

Do we need 2 fields?
```
"namespace" : "asic0",
"id" : "0"
```

I notice related scripts using both extensively, and deduce from one to another.

Reply:
I think you are trying to make the namespace name a bit more meaningful.

The idea I have is to change the namespace names from "0", "1", "2" .. --> to "namespace"0, "namespace"1, "namespace"2 etc ... 

Currently we use the string "asic0", "asic1", "asic2" etc . But I feel the prefix "asic" is not correct for a namespace name -- as tomorrow we could support more than 1 ASIC in the same namespace.

Can we keep it as "ns0", "ns1" etc ? Please share thoughts on the naming strategy... These names will be used as input string for "-namespace" parameter being introduced to sonic-utilities, config/show commands etc.

Reply:
Used the namespace name "asic0", "asic1" etc instead of "0", "1" etc